Jenkinsfile

```diff
@@ -0,0 +1,78 @@
+pipeline {
+    agent any
+    stages {
+        stage('Copy Base Build') {
+            when { not { environment name: 'JOPT_BASE_BUILD', value: '' } }
+            steps {
+                build job: 'Copy base build', parameters: [
+                    string(name: 'JOPT_BASE_BUILD', value: "${JOPT_BASE_BUILD}"),
+                    string(name: 'JOPT_UPLOAD_DIR', value: "${JOPT_UPLOAD_DIR}")
+                    ]
+            }
+        }
+        stage('Build') {
+            parallel {
+                stage('Linux build') {
+                    when { environment name: 'JOPT_RUN_LINUX64', value: 'true' }
+                    steps {
+                        script {
+                            def allParams = params.collect{
+                                if (it.value instanceof java.lang.Boolean) {
+                                    return booleanParam(name: it.key, value: Boolean.valueOf(it.value))
+                                } else {
+                                    return string(name: it.key, value: it.value)
+                                }
+                            }
```

Comment:
I assume there is no neat solution instead of repeating this - too sad :\

Reply:
None that I know of - which doesn't mean that one doesn't exist

Reply:
Now that I think about it, I didn't try just passing params as parameters - that might actually work. I'll try it tomorrow.

Reply:
Now of course that didn't work, as params is a map and parameters is a list :(